app/src/main/java/org/openobservatory/ooniprobe/activity/RunTestActivity.java

```diff
@@ -0,0 +1,236 @@
+                            final String appPackageName = getPackageName(); // getPackageName() from Context or Activity object
+                            try {
+                                startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + appPackageName)));
+                            } catch (android.content.ActivityNotFoundException anfe) {
+                                startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://play.google.com/store/apps/details?id=" + appPackageName)));
+                            }
+                        }
+                    });
+                }
+            }
+        }
+    }
+
+    public void configureScreen(String td, String ta){
+        TestData.getInstance(this, this).addObserver(this);
+         urls = (TextView) findViewById(R.id.urls);
+
+        title = (TextView) findViewById(R.id.run_test_message);
+        if (td != null)
+            title.setText(td);
+        else
+            title.setText(getString(R.string.run_test_message));
+
+        String test = NetworkMeasurement.getTestName(this, test_name);
+        testTitle = (TextView) findViewById(R.id.test_title);
+        testTitle.setText(test);
+
+        testImage = (ImageView) findViewById(R.id.test_logo);
+        testImage.setImageResource(NetworkMeasurement.getTestImageBig(test_name));
+
+        ArrayList<String> listItems = new ArrayList<>();
+        final ArrayList<String> urlItems = new ArrayList<>();
+        if (ta != null){
+            try {
+                JSONObject taJson = new JSONObject(ta);
+                JSONArray urls = taJson.getJSONArray("urls");
+                for (int i = 0; i < urls.length(); i++)
+                    urlItems.add(urls.getString(i));
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+            listItems.addAll(urlItems);
+        }
+
+        if (listItems.size() == 0 && test_name.equals("web_connectivity")){
+            listItems.add(getString(R.string.random_sampling_urls));
+            mUrlListAdapter = new UrlListAdapter(this, listItems, false);
+        }
+        else
+            mUrlListAdapter = new UrlListAdapter(this, listItems, true);
+
+        if (listItems.size() == 0)
+            urls.setVisibility(View.INVISIBLE);
+
+        runButton = (AppCompatButton) findViewById(R.id.run_test_button);
+        runButton.setOnClickListener(
+                new ImageButton.OnClickListener() {
+                    public void onClick(View v) {
+                        TestData.doNetworkMeasurements(getApplicationContext(), test_name, urlItems);
+                        Intent MainActivityIntent = new Intent(RunTestActivity.this, MainActivity.class);
+                        MainActivityIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+                        MainActivityIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
```

Comment:
It would be helpful to explain why the activity is being launched with these flags

Reply:
Also, I was wondering: the `MainActivity` is by default created with what flags? Does it matter that here we're launching it with non-default flags?

Reply:
^^^ @lorenzoPrimi

Reply:
Don't know, just copied from Android manual.

Reply:
@lorenzoPrimi can you then at least put a comment with a link to where you got the code snippet from?

Reply:
Here are the definitions:
https://developer.android.com/reference/android/content/Intent.html

I guess this one is the snippet
https://stackoverflow.com/questions/14332441/finish-current-activity-and-start-a-new-one

Reply:
@lorenzoPrimi excellent! Can you put that in a comment in the code?